python-sdk/kiota-gen.py

```diff
@@ -19,12 +25,13 @@ def generate_kiota_client_files(setup_kwargs):
         print("Unsupported operating system.")
         exit(1)
 
-    kiota_arch_name = KIOTA_ARCH_NAMES.get(platform.architecture()[0], None)
+    machine = platform.machine().lower()
+    kiota_arch_name = KIOTA_ARCH_NAMES.get(os_bits(machine), None)
     if kiota_arch_name is None:
         print("Unsupported architecture.")
         exit(1)
 
-    if platform.processor()[0] == "arm" and kiota_arch_name == "x64":
+    if machine.startswith("arm") and kiota_arch_name == "x64":
         kiota_arch_name = "arm64"
```

Comment:
I'd avoid this name patch, and possibly remove the bit-width references altogether.

Reply:
agree, removed 👍